### PR TITLE
PSR-3 adapter and writer

### PR DESCRIPTION
**Pull request by Xerkus**
This PR adds PSR-3 adapter for Zend\Log.
I had to add standalone adapter class as Psr logger interface can not be applied directly to Zend\Log\Logger since some methods are overlapping.

Also new Psr3 writer added to let Zend\Log write to any PSR-3 compatible loggers.

### CODE REVIEW

**Comment by gianarb** [comment]
I started this feature few time ago! :) Perfect :+1:

**Comment by weierophinney** [comment]
@Xerkus Could you do me a favor?
- `mkdir -p doc/book`
- `touch doc/book/psr3.md`

and then document both consuming `Zend\Log\Logger` in a PSR-3 implementation, as well as attaching a PSR-3 implementation as a log writer to `Zend\Log\Logger`? @GeeH will be porting docs here soon, but this way we have the feature documented once he does, and it's not something we forget later.

BTW, LOVE the PR - very clean, very straight-forward, and a no-brainer.

**Comment by Maks3w** [comment]
PSR-3 has a compliant test suite Could you add it?

File: test/Psr3LoggerAdapterTest.php [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+use Psr\Log\LogLevel;
+use Zend\Log\Logger;
+use Zend\Log\Psr3LoggerAdapter;
+
+/**
+ * @group Zend_Log
+ * @coversDefaultClass Zend\Log\Psr3LoggerAdapter
+ * @covers ::<!public>
+ */
+class Psr3LoggerAdapterTest extends TestCase
```
Comment (Maks3w) [inline test/Psr3LoggerAdapterTest.php]: extend from Psr\Log\Test\LoggerInterfaceTest

File: test/Writer/Psr3Test.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    /**
+     *
+     * @covers ::__construct
+     */
+    public function testConstructWithOptions()
+    {
+        $psr3Logger = $this->getMock('Psr\Log\LoggerInterface');
+        $formatter = new \Zend\Log\Formatter\Simple();
```
Comment (Maks3w) [inline test/Writer/Psr3Test.php]: Please import this classes

File: src/Writer/Psr3.php [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+namespace Zend\Log\Writer;
+
+use Psr\Log\LogLevel;
+use Psr\Log\LoggerAwareTrait as Psr3LoggerAwareTrait;
+use Psr\Log\LoggerInterface as Psr3LoggerInterface;
+use Psr\Log\NullLogger;
+use Zend\Log\Logger;
+
+class Psr3 extends AbstractWriter
```
Comment (Maks3w) [inline src/Writer/Psr3.php]: Please add a class block with a description for this class

**Comment by Xerkus** [comment]
will do
@weierophinney on a side note, this repo does not get irc notifications as it seems.

**Comment by Xerkus** [comment]
@Maks3w I'm not sure about implementing abstract logger interface test. In order for it to work i will need to manipulate log messages, which i find rather dirty.

I will need to do two things: 
- adapter maps log levels to Zend\Log priorities, i will need to map back and prepend log levels. Not much of a problem as it is 1 for 1 mapping.
- compat test expects and relies on placeholder substitution. This one is especially dirty since it will make related test pass. PSR-3 states placeholder substitution is optional and Zend\Log does not do it.

Any thoughts?

File: test/Writer/Psr3Test.php (hunk 2 above)
Comment (Xerkus) [inline test/Writer/Psr3Test.php]: I keep forgetting i can use `::class` now \o/

Comment (Maks3w) [inline test/Writer/Psr3Test.php]: require 5.6

**Comment by Xerkus** [comment]
Added docs.
I will greatly appreciate if my english will be fixed there.

**Comment by weierophinney** [comment]
Merged to develop for release with 2.6.0

**Pull request closed without merging**